app/src/main/java/org/sopt/and/SignUpActivity.kt

```diff
@@ -0,0 +1,217 @@
+                        containerColor = Color.DarkGray
+                    ),
+                    visualTransformation = if(passwordVisible) VisualTransformation.None else PasswordVisualTransformation(),
+                    trailingIcon = {
+                        Text(
+                            text = "show",
+                            color = Color.White,
+                            modifier = Modifier.clickable {
+                                passwordVisible = !passwordVisible
+                            }
+                        )
+                    }
+                )
+                Spacer(Modifier.height(10.dp))
+
+                Row {
+                    Image(
+                        painter = painterResource(R.drawable.baseline_info_outline_24),
+                        contentDescription = ""
+                    )
+                    Spacer(Modifier.width(5.dp))
+
+                    Text(
+                        text = stringResource(R.string.signup_password_explain),
+                        color = Color.Gray
+                    )
+                }
+            }
+
+
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .align(Alignment.BottomCenter)
+            ){
+                Button(
+                    onClick = {
+                        // 조건에 맞을 경우에만 화면 전환
+                        if (isAbleEmail(userId) && isAblePassword(userPassWord)){
+                            val resultIntent = Intent().apply {
+                                putExtra("userId", userId)
+                                putExtra("userPassWord", userPassWord)
+                            }
+                            activity.setResult(Activity.RESULT_OK, resultIntent)
+                            activity.finish()
+                            Toast.makeText(context,"회원가입 성공",Toast.LENGTH_SHORT).show()
+                        }else{
+                            Toast.makeText(context,"회원가입 실패",Toast.LENGTH_SHORT).show()
+                        }
+                    },
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .height(60.dp),
+                    colors = ButtonDefaults.buttonColors(
+                        containerColor = Color.LightGray.copy(alpha = 0.5f)
+                    ),
+                    shape = RectangleShape
+                ) {
+                    Text(
+                        text = stringResource(R.string.signup_button),
+                        color = Color.White
+                    )
+                }
+            }
+        }
+    }
+}
+
+fun isAbleEmail(email: String): Boolean{
+    return android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
+}
+
+fun isAblePassword(password: String): Boolean{
+    val pattern = "^(?=.*[A-Za-z])(?=.*\\d)(?=.*[@\$!%*?&])[A-Za-z\\d@\$!%*?&]{8,20}\$"
+    return Regex(pattern).matches(password)
```

Comment:
정규표현식 객체는 생성비용이 커서 파일 내에 변수로 한번만 만들어서 사용하는것이 권장됩니다!

```suggestion
val PASSWORD_REGEX = Regex("^(?=.*[A-Za-z])(?=.*\\d)(?=.*[@\$!%*?&])[A-Za-z\\d@\$!%*?&]{8,20}\$")
fun isAblePassword(password: String): Boolean{
    return PASSWORD_REGEX.matches(password)
}
```

Reply:
앗 넵!